### PR TITLE
[onert] Separate NativeTensor of TensorRegistry in train backend

### PR DESCRIPTION
**Pull request by ragmani**
This commit separates NativeTensor of TensorRegistry into NonConstTensor and TrainableTensor in train backend.

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

**Comment by glistening** [comment]
@ragmani As I understand, native tensor is about whether the tensor is created from its backend or other backend. I don't have idea how and why this concept is related to trainable tensor. Could you please explain the background or design behind this PR?

**Comment by ragmani** [comment]
@glistening 
`TrainableTensor` will be introduced at #11007. And within train backend, we can handle `Tensor` for non constants and `TrainableTensor` for constants(like weights) separately.
Also, outside of train backend we can get native tensors by using `getNativeITensor()`.

File: runtime/onert/backend/train/TensorRegistry.h [hunk 1]
```diff
@@ -82,7 +85,7 @@ class TensorRegistry : public PortableTensorRegistryTemplate<Tensor>
     return PortableTensorRegistryTemplate<Tensor>::setMigrantTensor(ind, tensor);
   }
 
-  void setNativeTensor(const ir::OperandIndex &ind, std::unique_ptr<Tensor> tensor)
+  void setNonConstTensor(const ir::OperandIndex &ind, std::unique_ptr<Tensor> tensor)
```
Comment (glistening) [inline runtime/onert/backend/train/TensorRegistry.h:88]: Please update exception message at line 94.

Also, before setting non-const tensor, it checks `_trainable`.
Does every non-const tensor is in _trainable?
I guess there has been non-const tensors before trainable tensor is introduced.

Comment (ragmani) [reply runtime/onert/backend/train/TensorRegistry.h:88]: > Please update exception message at line 94.

OK, I will change "native" to "non const"

> Also, before setting non-const tensor, it checks _trainable.
> Does every non-const tensor is in _trainable?

No, It checks whether the index is valid. Every const tensor is in `_trainable`.

**Review by glistening — APPROVED** [review]

**Pull request merged**